### ISSUE TITLE
Incompatibility with Dash to Dock extension?

### ISSUE THREAD

**Issue by heftig** [franziskuskiefer/app-keys-gnome-shell-extension#17]
The shortcuts often don't work, and the shell produces this error when a shortcut is attempted:

```
(gnome-shell:6310): St-CRITICAL **: st_widget_get_theme_node called on the widget [0xa94bd20 ShellGenericContainer "dash"] which is not in the stage.
(gnome-shell:6310): Gjs-WARNING **: JS ERROR: Error: can't convert "x" to an integer
myAppIcon<.updateIconGeometry@/home/jan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1358
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
myDash<._updateAppIconsGeometry/<@/home/jan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:773
myDash<._updateAppIconsGeometry@/home/jan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:772
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
myDash<._redisplay@/home/jan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1066
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
AppKeys.prototype.clickClosure/<@/home/jan/.local/share/gnome-shell/extensions/unitylike-hotkey@webgyerek.net/extension.js:41
```

Shell 3.16.3
Dash to Dock issue: https://github.com/micheleg/dash-to-dock/issues/198

**Comment by franziskuskiefer** [comment]
Looks like the `Dash._redisplay()` is not working with DtD anymore. I'll try to have a look at this but I'm somewhat confused by `can't convert "x" to an integer`.

**Comment by micheleg** [comment]
Hi, I'm the author of dash-to-dock. That error is most likely unrelated. If you elaborate on why you need to call the _redisplay method I might help understanding what could be wrong.

**Comment by franziskuskiefer** [comment]
There used to be a bug that required this (this is from before I took over this extension, so I'm not actually sure what the problem there was). I'll live locally without it for a while to see if I encounter any problems and might just remove it if not.

@heftig could you comment line 41 in extension.js locally and let me know if that fixes the problem and works in general?

**Comment by heftig** [comment]
I've commented it out now. We'll see.

**Comment by ainola** [comment]
@heftig, any news? I've been bitten by this, too. it appears that commenting out the line and restarting gnome-shell seems to have fixed the issue. I'll report back if I hit any walls.

**Comment by franziskuskiefer** [comment]
I've currently limited internet access, but since it seems to be fine
removing the line that causes trouble I'll do that and upload a new version
as soon as I have internet again.

**Comment by ainola** [comment]
Yep, everything has been just fine on two computers.

**Comment by franziskuskiefer** [comment]
fixed in d505bc8

**Comment by franziskuskiefer** [comment]
is now updated on the extension website as well
